hilti/runtime/src/tests/bytes.cc

```diff
@@ -352,20 +361,20 @@ TEST_CASE("toInt") {
     }
 
     SUBCASE("with byte order") {
-        CHECK_EQ("100"_b.toInt(ByteOrder::Big), 3223600);
-        CHECK_EQ("100"_b.toInt(ByteOrder::Network), 3223600);
-        CHECK_EQ("100"_b.toInt(ByteOrder::Little), 3158065);
-        if ( systemByteOrder() == ByteOrder::Little ) {
-            CHECK_EQ("100"_b.toInt(ByteOrder::Host), 3158065);
+        CHECK_EQ("100"_b.toInt(ByteOrder(ByteOrder::Big)), 3223600);
```

Comment:
The issue in this particular location is that `toInt` is overloaded to accept integers. `ByteOrder::Value` is a C-style unscoped enum which is also just an integer. This makes it I think impossible to somehow infer that we want a "proper enum" here. We also have this duplication for e.g., `toString` for enum values which I think have a similar issue.

It might be possible to avoid this duplication by generating two types for a HILTI enum, a scoped enum like we had previously and a class type with proper rt-type semantics, e.g., here an `enum class ByteOrder` and a `struct ByteOrder_t`. We could then add implicit conversions between the enum and the class type, and route any operations on the enum through the class. That would allow us to avoid the name duplication here and elsewhere, but be pretty heavy-handed.

Reply:
Made the change using ADL you suggested.